### PR TITLE
Fix Android engine flags defaulting to true for malformed values

### LINKED ISSUE #184581: Parsing of android flutter shell args currently defaults to true
This was changed with 
https://github.com/flutter/flutter/pull/182522/changes
See https://github.com/flutter/flutter/pull/182522#discussion_r3033338235

We should give the flag definition a place for a default value, and then give each flag the default it had before, to restore the old behavior.

Note that we currently are checking if the key is present
https://github.com/flutter/flutter/blob/master/engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java#L314

So this won't affect most cases, but the behavior for a flag with a typo in the value (`Fasle`), or no value, changed from defaulting from false to true for many of these flags.

### PR DESCRIPTION
**Pull request by realmeylisdev**
## Summary
- `FlutterLoader` parses Android manifest metadata for engine shell flags, but `getBoolean(metadataKey, true)` defaults to `true` when the value is missing or unparseable — accidentally enabling flags that should be off
- Change the default from `true` to `false`, matching the pre-#182522 per-flag behavior where flags were disabled unless explicitly enabled

## Changes
- `engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java`: Changed `applicationMetaData.getBoolean(metadataKey, true)` to `getBoolean(metadataKey, false)`

Fixes flutter/flutter#184581

### CODE REVIEW

File: engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java [hunk 1]
```diff
@@ -396,8 +396,9 @@ void ensureInitializationComplete(
           }
 
           // Check if a boolean value is specified and if so, use it to determine if the
-          // flags should be added. If not, assume the flag is meant to be added.
-          if (applicationMetaData.getBoolean(metadataKey, true)) {
+          // flags should be added. If the value is missing or unparseable, default to
+          // false (disabled), matching the pre-#182522 per-flag behavior.
```
[the remark this answers is not in the document]
Comment (camsim99) [reply engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java]: I like this edit.

**Review by camsim99 — COMMENTED** [review]
This approach LGTM! This does not directly follow the guidance in https://github.com/flutter/flutter/issues/184581 about providing default values for each flag, but I think defaulting to false in the case that true is not explicitly specified is more straightforward.

@gmackall Any concerns with just defaulting to false instead? I think the default value proposal is more of a nice to have, but this fix is good for now. I believe that all of the flags should default to false at this point in the logic if not specified in the manifest; I realize I left this bug because I switched from searching through the manifest for flags to looping through all of the flags and using a check to see if they are in the manifest.

@realmeylisdev Some requests if we move forward with this, though:

1. We will need a test for this in `FlutterLoaderTest.java`.
2. We should edit https://github.com/flutter/flutter/blob/8422f21c7a862e1ee3b16fb087772ed1b60d1523/docs/engine/Flutter-Android-Engine-Flags.md?plain=1#L96 as well to clarify that `true` needs to be specified for boolean flags.

File: engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (realmeylisdev) [reply engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java]: Done — removed the PR number reference. Comment now reads: *"false (disabled) to ensure flags are only enabled when explicitly requested."*

[the remark this answers is not in the document]
Comment (realmeylisdev) [reply engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java]: Done — addressed both:

1. **Test added** — `itDoesNotSetBooleanFlagWithMalformedOrMissingValue` covers both a typo string (`"Fasle"`) and a `null` value. Also updated all 20 existing boolean flag tests that were passing `null` to pass explicit `true` instead, since they would have broken with the new default.
2. **Docs updated** — Replaced the "omit a value = true" section in `Flutter-Android-Engine-Flags.md` with guidance that `android:value="true"` must be explicitly set.

**Review by camsim99 — DISMISSED** [review]
This LGTM, thanks for the quick fixes! We just need a second review.

**Review by gmackall — DISMISSED** [review]
AFAICT all the values which had defaults defaulted to false before so this seems fine to me. I think we can leave adding a default attribute for the future when we have a flag which we desire to have non false default.

Thanks for the contribution!

**Comment by camsim99** [comment]
This appears to be blocked by https://github.com/flutter/flutter/issues/184689.

**Review by camsim99 — APPROVED** [review]
Re-approving after fixing format.

**Review by gmackall — APPROVED** [review]

**Review by camsim99 — APPROVED** [review]

**Review by jesswrd — APPROVED** [review]
